packages/actor-query-operation-extend/lib/ActorQueryOperationExtend.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  ActorQueryOperation, ActorQueryOperationTypedMediated, Bindings,
+  IActorQueryOperationOutput,
+  IActorQueryOperationOutputBindings,
+  IActorQueryOperationTypedMediatedArgs,
+} from "@comunica/bus-query-operation";
+import { BindingsStream } from "@comunica/bus-query-operation";
+import { ActionContext, IActorTest } from "@comunica/core";
+import { termToString } from 'rdf-string';
+import { Algebra } from "sparqlalgebrajs";
+import { AsyncEvaluator, ExpressionError } from "sparqlee";
+
+/**
+ * A comunica Extend Query Operation Actor.
+ *
+ * See https://www.w3.org/TR/sparql11-query/#sparqlAlgebra;
+ */
+export class ActorQueryOperationExtend extends ActorQueryOperationTypedMediated<Algebra.Extend> {
+
+  constructor(args: IActorQueryOperationTypedMediatedArgs) {
+    super(args, 'extend');
+  }
+
+  public async testOperation(pattern: Algebra.Extend, context: ActionContext): Promise<IActorTest> {
+    // Will throw error for unsupported opperations
+    const _ = new AsyncEvaluator(pattern.expression);
+    return true;
+  }
+
+  public async runOperation(pattern: Algebra.Extend, context: ActionContext)
+    : Promise<IActorQueryOperationOutputBindings> {
+
+    const { expression, input, variable } = pattern;
+
+    const output: IActorQueryOperationOutputBindings = ActorQueryOperation.getSafeBindings(
+      await this.mediatorQueryOperation.mediate({ operation: input, context }));
+
+    const extendKey = termToString(variable);
+    const evaluator = new AsyncEvaluator(expression);
+
+    // Transform the stream by extending each Bindings with the expression result
+    const transform = async (bindings: Bindings, next: any) => {
+      try {
+        const result = await evaluator.evaluate(bindings);
+        const extended = bindings.set(extendKey, result); // Extend is undefined when the key exists.
```

Comment:
Is this comment a TODO? Should this be _Extend is undefined when the **does not** key exists_?

Reply:
Actually not, when you extend with a variable name already present in the bindings, the result of extend is undefined by the spec. In this case, it'll just overwrite. I'll rephrase the comment.